spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+    {
+        if (providerName == null || action CALL_METHOD(providerName, "length", []) == 0)
+            _throwIAE();
+
+        val providersList: array<Provider> = action CALL_METHOD(null as Security, "getProviders", []);
+        val providersListLength: int = action ARRAY_SIZE(providersList);
+        var resultProvider: Provider = null;
+        var resultAlgorithm: String = null;
+
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, providersListLength, +1,
+            findAlgorithmByAllProvidersByName_loop(i, providersList, _algorithm, resultProvider, resultAlgorithm, providerName)
+        );
+
+        if (resultProvider == null)
+            _throwNSPE();
+
+        if (resultAlgorithm == null)
+            _throwNSAE();
+
+        val isDefaultProvider: boolean = _isDefaultProvider(resultProvider);
+
+        result = new SecureRandomAutomaton(state = Initialized,
+            provider = resultProvider,
+            algorithm = resultAlgorithm,
+            defaultProvider = isDefaultProvider,
+        );
+    }
+
+
+    @Phantom proc findAlgorithmByAllProvidersByName_loop (i: int, providersList: array<Provider>, algorithm: String, resultProvider: Provider, resultAlgorithm: String, providerName: String): void
+    {
+        val curProvider: Provider = providersList[i];
+        val curProviderName: String = action CALL_METHOD(curProvider, "getName", []);
+        if (action OBJECT_EQUALS(curProviderName, providerName))
+        {
+            val services: Set = action CALL_METHOD(curProvider, "getServices", []);
+            val iter: Iterator = action CALL_METHOD(services, "iterator", []);
+            resultProvider = curProvider;
+
+            action LOOP_WHILE(
+                action CALL_METHOD(iter, "hasNext", []),
+                findAlgorithm_loop(iter, curProvider, algorithm, resultAlgorithm)
+            );
+        }
+    }
+
+
+    @throws(["java.security.NoSuchAlgorithmException"])
+    @static fun *.getInstanceStrong (): SecureRandom
+    {
+        // #note this property can be disabled with SecurityManager. We assume that this situation is impossible,
+        // because we can't check enable/disable property status. And we can't invoke "AccessController.doPrivileged" method like in source code;
+
+        val propertyName: String = "securerandom.strongAlgorithms";
+        val property: String = action CALL_METHOD(null as Security, "getProperty", [propertyName]);
+
+        if (property == null)
+            _throwNSAE();
+
+        val propertyLength: int = action CALL_METHOD(property, "length", []);
+        if (propertyLength == 0)
+            _throwNSAE();
+
+        val algorithms: array<String> = action CALL_METHOD(property, "split", [","]);
+        val algorithmsLength: int = action ARRAY_SIZE(algorithms);
+
+        var j: int = 0;
+        action LOOP_FOR(
+            j, 0, algorithmsLength, +1,
+            findAlg_loop(j, algorithms, result)
```

Comment:
@dog-m 
This loop was deleted, because were used:

```
action SYMBOLIC("java.security.Provider");
action SYMBOLIC("java.lang.String");
```

temporary decision;